pandas/tests/frame/test_join.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+
+import pandas as pd
+
+from pandas.tests.frame.common import TestData
+
+import pandas.util.testing as tm
+from pandas.util.testing import (assertRaisesRegexp,
+                                 assert_frame_equal)
+
+
+class TestDataFrameJoin(tm.TestCase, TestData):
```

Comment:
since you are writing new tests, pls use the pytest style. don't inherit from ``tm.TestCase``), so don't use ``self.`` for anything testing related

e.g. just use

``assert_frame_equal`` and such